src/frontend/src/handler/alter_source_column.rs

```diff
@@ -0,0 +1,179 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pgwire::pg_response::{PgResponse, StatementType};
+use risingwave_common::catalog::ColumnId;
+use risingwave_common::error::{ErrorCode, Result, RwError};
+use risingwave_connector::source::{SourceEncode, SourceStruct};
+use risingwave_source::source_desc::extract_source_struct;
+use risingwave_sqlparser::ast::{AlterSourceOperation, ObjectName};
+
+use super::create_table::bind_sql_columns;
+use super::util::alter_definition_add_column;
+use super::{HandlerArgs, RwPgResponse};
+use crate::catalog::root_catalog::SchemaPath;
+use crate::Binder;
+
+// Note for future drop column:
+// 1. Dependencies of generated columns
+
+/// Handle `ALTER TABLE [ADD] COLUMN` statements.
+pub async fn handle_alter_source_column(
+    handler_args: HandlerArgs,
+    source_name: ObjectName,
+    operation: AlterSourceOperation,
+) -> Result<RwPgResponse> {
+    // Get original definition
+    let session = handler_args.session;
+    let db_name = session.database();
+    let (schema_name, real_source_name) =
+        Binder::resolve_schema_qualified_name(db_name, source_name.clone())?;
+    let search_path = session.config().get_search_path();
+    let user_name = &session.auth_context().user_name;
+
+    let schema_path = SchemaPath::new(schema_name.as_deref(), &search_path, user_name);
+
+    let (db_id, schema_id, mut catalog) = {
+        let reader = session.env().catalog_reader().read_guard();
+        let (source, schema_name) =
+            reader.get_source_by_name(db_name, schema_path, &real_source_name)?;
+        println!("Schema_name: {schema_name}");
+        let db = reader.get_database_by_name(db_name)?;
+        let schema = db.get_schema_by_name(schema_name).unwrap();
+
+        session.check_privilege_for_drop_alter(schema_name, &**source)?;
+
+        (db.id(), schema.id(), (**source).clone())
+    };
+
+    // Currently only allow source without schema registry
+    let SourceStruct { encode, .. } = extract_source_struct(&catalog.info)?;
+    match encode {
+        SourceEncode::Avro | SourceEncode::Protobuf => {
+            return Err(RwError::from(ErrorCode::NotImplemented(
+                "Alter source with schema registry".into(),
+                None.into(),
+            )));
+        }
+        SourceEncode::Invalid | SourceEncode::Native => {
+            return Err(RwError::from(ErrorCode::NotSupported(
+                format!("Alter source with encode {:?}", encode),
+                "Alter source with encode JSON | BYTES | CSV".into(),
+            )));
+        }
+        _ => {}
+    }
+
+    let columns = &mut catalog.columns;
+    match operation {
+        AlterSourceOperation::AddColumn { column_def } => {
+            let new_column_name = column_def.name.real_value();
+            if columns
+                .iter()
+                .any(|c| c.column_desc.name.to_lowercase() == new_column_name)
```

Comment:
```suggestion
                .any(|c| c.column_desc.name == new_column_name)
```
I think we should leave all things to the `real_value()`